### PR TITLE
Ok 5 report reviews

### PR DESCRIPTION
**Pull request by dufourmax1738**
# Jira Link:

[[Jira: ]()](https://champlainsaintlambert.atlassian.net/jira/software/projects/OK/boards/55?selectedIssue=OK-5)

# Context

Users can now report content

# Changes

- added report popup
- report popup called when clicking report
- added firebase function to check if user already reported a review
- added firebase function to add report
- added report type

# Before and After UI (Required for UI-Impacting PRs) 

[After: ]()
image

### CODE REVIEW

**Review by ClementCadieux — APPROVED** [review]
Everything looks good, can you just explain what the change to the test had to be?

**Comment by dufourmax1738** [comment]
> Everything looks good, can you just explain what the change to the test had to be?

Yup! The problem was the test assumes there is always a dropdown button even when logged out however this does not make sense. Someone who is not logged in should not be able to report a review because this could be exploited and spammed. Therefore I removed the dropdown button when a user is not logged in which means that cypress should not be able to see it. I changed the test so that it checks to make sure that there is no drop-down button when logged out.

:shipit:

File: lib/firebase.ts [hunk 1]
```diff
@@ -2,8 +2,21 @@ import firebaseConfig from "./firebaseConfig";
...
+    doc,
+    getDoc,
+    getDocs,
+    limit,
+    orderBy,
+    query,
+    serverTimestamp,
+    setDoc,
+    where
+} from "@firebase/firestore";
```
Comment (DuncanBH) [inline lib/firebase.ts:17]: what did you do to this poor import???

**Review by DuncanBH — CHANGES_REQUESTED** [review]

File: lib/firebase.ts (hunk 1 above)
Comment (dufourmax1738) [reply lib/firebase.ts:17]: i have no idea i didnt touch this

**Review by DuncanBH — APPROVED** [review]

**Pull request merged**